lunabot_nav/src/lunabot_nav/dstar.py

```diff
@@ -1,92 +1,105 @@
+        """
 
-        self.node_values_list = maxsize * np.ones(
-            (init_map.shape[0], init_map.shape[1], 2)
-        )  # 2d Array of nodes: each value is [Distance (g), Estimate (rhs)]
+        self.node_queue: PriorityQueue = PriorityQueue()
+        self.km: float = 0.0
 
-        self.x_offset = x_offset
-        self.y_offset = y_offset
+        # What number on the map corresponds to occupied
+        self.OCCUPANCY_THRESHOLD = 50  
 
-        self.current_map = init_map
+         # faster than computing sqrt 2 over and over
+        self.root2 = np.sqrt(2)
 
-        self.unexpanded_map = init_map
+        # 2d Array of nodes that corresponds to the map: each value is [Distance (g), Estimate (rhs)]
+        self.node_values_list: np.ndarray[float] = maxsize * np.ones((init_map.shape[0], init_map.shape[1], 2))  
 
-        self.res = resolution
+        # Real-world location of the 0,0 in the grid (not including buffer in the map)
+        self.x_offset: float = x_offset 
+        self.y_offset: float = y_offset
 
-        goal = self.convertToGrid(goal)
+        # The amount of buffer on any side of the map
+        self.buffer_offset_left: int = 0
+        self.buffer_offset_up: int = 0
+        self.buffer_offset_right: int = 0
+        self.buffer_offset_down: int = 0
 
-        self.node_values_list[goal[0], goal[1], 1] = 0
+        # 2d array occupancy map: Occupancy probabilities from [0 to 100].  Unknown is -1.
+        self.current_map: np.ndarray[int] = init_map
 
-        start = self.convertToGrid(start)
+        self.resolution: float = resolution # meters per grid cell
 
-        self.current_node = start
-        self.prev_node = start
+        # Save the real-world position of the goal
+        self.real_goal = goal 
 
-        self.goal = goal
+        self.goal: list[int] = self.convert_to_grid(goal) # Convert the goal to grid coordinates initially
+        self.buffer_map_for_goal()                        # Add a buffer to the map so we can plan to the goal when it is outside the map
+        self.goal = self.convert_to_grid(goal)            # Reconvert given the new buffer
 
-        self.radius = radius
+        self.node_values_list[self.goal[0], self.goal[1], 1] = 0 # Set the estimate (rhs) value of the goal to 0
 
-        self.needs_new_path = (
-            True  # updates when map changes to know when to create new path
-        )
+        start = self.convert_to_grid(start) # Convert the start to grid coordinates
 
-        self.insert(goal, self.calculate_key(goal, False))
+        self.current_node: list[int] = start
+        self.prev_node: list[int] = start
 
-    # test
+        # Insert the goal node into the priority queue
+        self.insert(self.goal, self.calculate_key(self.goal))
 
-    # Update Pose (current position)
     def update_position(self, coords):
-        position = self.convertToGrid(coords)
+        """
```

Comment:
You only expand the buffer at the beginning for the goal. What happens if the robot traverses into a space where the buffer needs to be expanded further? I don't think thats addressed in these changes

Reply:
Nevermind, didn't see the buffer gets updated in `update_map`. I'm still a bit concerned about timing w/ `update_pos` (if `update_pos` gets called before `update_map`, then may run into an issue where you try setting the pos as a negative grid index)

Reply:
Looked at this further / discussed, it should be ok